### PR TITLE
Updates to the trainer to work with MSE, and perform train/test split when test set isn't provided

### PR DESCRIPTION
**Pull request by PraneetNeuro**
This PR comprises of the following updates : 
 - Updated DSpritesConfig to reflect the dataset
 - dsprites dataset object updated to support indexing of a target class 
 - trainer updated to work with MSE, and perform train/test split in case of test split not being provided

### CODE REVIEW

**Comment by soniajoseph** [comment]
Updates to trainer great! Remove DSpriteConfig and dataloader?

**Comment by PraneetNeuro** [comment]
The PR now comprises only updates to the trainer.

**Comment by soniajoseph** [comment]
Excellent ty
